### PR TITLE
Explore in Datasette

### PR DESCRIPTION
**Pull request by garethrees**
Related to https://github.com/mysociety/alaveteli/issues/7027

Screenshot 2023-10-17 at 10 22 26

With custom URL:

Screenshot 2023-10-19 at 17 56 08

CORS headers:

Screenshot 2023-10-17 at 10 23 03

### CODE REVIEW

File: spec/helpers/datasette_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        it { is_expected.to include("https://d.example.com/?csv=http") }
+      end
+    end
+
+    context 'the attachment is not a CSV' do
+      let(:attachment) { FactoryBot.build(:rtf_attachment) }
+      it { is_expected.to be_nil }
+    end
+  end
```
Comment (garethrees) [inline spec/helpers/datasette_helper_spec.rb:39]: need a spec for hidden

File: app/helpers/datasette_helper.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  include InfoRequestHelper
+
+  mattr_accessor :datasette_url, default: 'https://lite.datasette.io/'
+
+  def explore_in_datasette(attachment)
+    return unless explorable_in_datasette?(attachment)
+
+    link_to "#{datasette_url}?csv=#{attachment_url(attachment)}" do
+      _('Explore in Datasette')
+    end
```
Comment (gbp) [inline app/helpers/datasette_helper.rb:12]: Could break depending on what `datasette_url` is set to, with #8030 we could do:

```suggestion
    datasette_attachment_url = add_query_params_to_url(
      datasette_url, csv: attachment_url(attachment)
    )

    link_to _('Explore in Datasette'), datasette_attachment_url
```

**Review by gbp — CHANGES_REQUESTED** [review]
Couple of RuboCop warnings and one small improvement.

File: app/helpers/datasette_helper.rb (hunk 2 above)
Comment (garethrees) [reply app/helpers/datasette_helper.rb:12]: Nice will incorporate https://github.com/mysociety/alaveteli/pull/8030 once merged

Comment (gbp) [reply app/helpers/datasette_helper.rb:12]: Merged

**Review by gbp — APPROVED** [review]

**Pull request merged**